src-tests/ConfigProviderTest.php

```diff
@@ -36,6 +36,14 @@ public function testShouldMakeConfigOptionsAccessibleDirectlyThroughConfigProvid
         $this->assertEquals('http://server.tld:port', ConfigProvider::getInstance()->serverUrl);
     }
 
+    public function testShouldDetectEmptyConfigOption()
+    {
+        ConfigHelper::setEnvironmentVariables($this->environmentVariables);
+
+        $this->assertFalse(empty(ConfigProvider::getInstance()->serverUrl));
+        $this->assertTrue(empty(ConfigProvider::getInstance()->capability));
```

Comment:
No, it won't detect if the __isset method is working properly, because PHPUnit will dereference the value by copying it into custom variable and then running empty() on it. So the test will still pass even if the _isset method is missing. You actaully really need to test directly either empty(), isset() or the __isset call...